### ISSUE TITLE
Make Request.Filters public

### ISSUE THREAD

**Issue by Jericho** [Pathoschild/FluentHttpClient#42]
This is related to discussion #38, but I thought it warranted a separate ticket.

Is there a reason `Request.Filters` is private instead of being public? I'm asking because I have a custom error handler that I need for all the calls to the vendor API except for one specific request where I need to remove this error handler. I still need the default error handler in case the vendor API throws  HTTP 500 or something, but I need to turn off my custom logic. Ideally, I would like to be able to write:

```CSharp
var request = _client
	.GetAsync(endpoint)
	.WithCancellationToken(cancellationToken);

request.Filters.Remove<AcmeErrorHandler>();

var myObject= await request.As<AcmeObject>().ConfigureAwait(false);
```

I propose the following:

1. Add `Filters` to the `IRequest` interface
2. Change `Request.Filters` to `public`

This would be similar to the `IRequest.Formatters` public property.

Please note that the `WithHttpErrorAsException` discussed in #38 would not help me in this situation since I want the default error handler to be enabled. It's just my custom handler that needs to be disabled.

**Comment by Pathoschild** [comment]
Hmm. It was deliberately internal since I considered it an implementation detail, but that's a valid use case. I'm okay with exposing it.

**Comment by Jericho** [comment]
Excellent, I'll submit PR

**Comment by Pathoschild** [comment]
Merged into `develop` for the upcoming 3.0 release. Thanks!
